Add sfx wave 2

rgarner opened the pull request:
Add and wire up new SFX:

- Break block
- Head against unbreakable block
- Spring

## Notes

- I'm not sure that the `box.breaking_right_now` setter is the best way of transmitting state here. It only exists to say "this box is breaking right now, and as such it shouldn't have a second sound triggered if you hit it again". Could just be naming; if we changed the `breaking` name to `breakable` then `breaking` could replace `breaking_right_now`
- I'm not sure about abusing the `landOnPlatform` interaction (again, might just be naming) to play the "bump head" sound.
- This is late night work buggering about with Phaser for the first time and as such you should probably treat anything other than the sounds themselves with a hazmat suit. I've put the sounds and the wiring in different commits in case you feel like doing things differently.

rgarner committed:
Add new SFX

- Break block
- Head against unbreakable block
- Spring

rgarner committed:
Wire up break/spring/head_block SFX

rgarner committed:
Fix audio/sfx Makefile

Without ND_SOURCE_WAVS on the % rule, we didn't know what to make when
ND_SOURCE_WAVS came from somewhere other than . – fix that so that
make from the external Logic location works properly

AJFaraday commented:
Thanks for the sounds!

I'll take a good luck when I next play with the code but these are my first thoughts.

* I think you're right about breaking/breakable. If there's a distinction, it's the current `breaking` attribute that should change.
* landOnPlatform could be renamed, too. Perhaps impactPlatform or meetPlatform. I'll have a think. 
* Changes look like they shouldn't effect the rest of the game, of course I'll check it out.

rgarner committed:
Rename breaking_box to breakable_box

Stop using `breaking` to refer to a box that is `breakable`. Keep
`breaking_right_now` for now.

rgarner committed:
Rename `breaking_right_now` -> `breaking`

Now we've removed any reference to `breaking_box` in favour of
`breakable_box`, we can use `if(platform.breaking)` to mean "is in
the process of breaking"

rgarner commented:
I've done the `breaking` -> `breakable` change over two commits. I like `meetPlatform`, also `collideWithPlatform` is in keeping with Phaser nomenclature.
